### ISSUE TITLE
[e2e]: docker-based workflow_dispatch main default slsa3 

### ISSUE THREAD

**Issue by ianlewis** [slsa-framework/slsa-github-generator#1809]
Repo: https://github.com/slsa-framework/example-package/tree/main
Run: https://github.com/slsa-framework/example-package/actions/runs/4401076291
Workflow file: https://github.com/slsa-framework/example-package/tree/main/.github/workflows/e2e.docker-based.workflow_dispatch.main.default.slsa3.yml
Workflow runs: https://github.com/slsa-framework/example-package/actions/workflows/e2e.docker-based.workflow_dispatch.main.default.slsa3.yml
Trigger: workflow_dispatch
Branch: main
Date: Mon Mar 13 03:23:32 UTC 2023

**Comment by ianlewis** [comment]
Is the number in the `RUNNER_NAME` some kind of version number? I suppose this test would then fail whenever the version is updated?

```
 GitHub Actions 9 == GitHub Actions 4 :: .predicate.buildDefinition.systemParameters.RUNNER_NAME should be GitHub Actions 4
```

**Comment by ianlewis** [comment]
The check looks like it should be ok. It seems like perhaps a workflow could run jobs on GitHub Actions runners that have different versions and that `RUNNER_NAME` could be different from job to job? That brings into question some of our assumptions about values being the same between the `setup-token` job and the job that runs the delegator workflow.

**Comment by asraa** [comment]
> It seems like perhaps a workflow could run jobs on GitHub Actions runners that have different versions and that RUNNER_NAME could be different from job to job?

Yes, I keep hitting this in my workflows: see https://github.com/slsa-framework/slsa-github-generator/pull/1785

I havent yet seen it in any of the delegator workflows, but maybe we don't have that testing set up yet -- our verify-token (which does check RUNNER_NAME) runs in the same job where it was created: https://github.com/slsa-framework/slsa-github-generator/blob/8762ebb0d785f04513ff2900d2a5c7e771cf81b3/.github/workflows/e2e.verify-token.schedule.yml#L21-L48

Realistically, we should separate the jobs out, to mimic how they would be called in a TRW, IMO cc @laurentsimon

**Comment by asraa** [comment]
RUNNER_NAME is verified in the common build definition scripts: https://github.com/slsa-framework/slsa-github-generator/blob/4df91f822551fe7916cc7ebc513c8f429fd26181/.github/workflows/scripts/e2e-verify.common.sh#L90

(but yes, wouldn't have seen it in actual delegator runs!)

**Comment by ianlewis** [comment]
> Realistically, we should separate the jobs out, to mimic how they would be called in a TRW, IMO cc @laurentsimon

Right, we were running both in the same job in the e2e tests... +1

**Comment by laurentsimon** [comment]
I think we should remove these field from the predicate since they represent the values of the caller, not the value where the build takes place https://github.com/slsa-framework/slsa-github-generator/issues/1817. Wdut?

**Comment by ianlewis** [comment]
> I think we should remove these field from the predicate since they represent the values of the caller, not the value where the build takes place #1817. Wdut?

Short term, we should remove it since it's potentially inaccurate info. I'll write on the other bug about ideas to potentially capture that info.

**Comment by ianlewis** [comment]
Repo: https://github.com/slsa-framework/example-package/tree/main
Run: https://github.com/slsa-framework/example-package/actions/runs/4411875852
Workflow file: https://github.com/slsa-framework/example-package/tree/main/.github/workflows/e2e.docker-based.workflow_dispatch.main.default.slsa3.yml
Workflow runs: https://github.com/slsa-framework/example-package/actions/workflows/e2e.docker-based.workflow_dispatch.main.default.slsa3.yml
Trigger: workflow_dispatch
Branch: main
Date: Tue Mar 14 03:24:37 UTC 2023
